buckets/archive/tracker.go

```diff
@@ -0,0 +1,265 @@
+							log.Errorf("rescheduling tracked archive: %s", err)
+						}
+					}(a)
+				}
+				wg.Wait()
+			}
+
+		}
+	}
+}
+
+func (t *Tracker) Track(ctx context.Context, dbID thread.ID, dbToken thread.Token, bucketKey string, jid ffs.JobID, bucketRoot cid.Cid) error {
+	if err := t.colls.ArchiveTracking.Create(ctx, dbID, dbToken, bucketKey, jid, bucketRoot); err != nil {
+		return fmt.Errorf("saving tracking information: %s", err)
+	}
+	return nil
+}
+
+// trackArchiveProgress queries the current archive status.
+// If a fatal error in tracking happens, it will return an error, which indicates the archive should be untracked.
+// If the archive didn't reach a final status yet, or a possibly recoverable error (by retrying) happens, it will return (true, "retry cause", nil).
+// If the archive reach final status, it will return (false, "", nil) and the tracking can be considered done.
+func (t *Tracker) trackArchiveProgress(ctx context.Context, buckKey string, dbID thread.ID, dbToken thread.Token, jid ffs.JobID, bucketRoot cid.Cid) (bool, string, error) {
+	log.Infof("querying archive status of job %s", jid)
+	defer log.Infof("finished querying archive status of job %s", jid)
+	ffsi, err := t.colls.FFSInstances.Get(ctx, buckKey)
+	if err != nil {
+		return true, fmt.Sprintf("getting instance data: %s", err), nil
+	}
+	// Step 1: watch for the Job status, and keep updating on Mongo until
+	// we're in a final state.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	ctx = context.WithValue(ctx, powc.AuthKey, ffsi.FFSToken)
+	ch := make(chan powc.JobEvent, 1)
+	if err := t.pgClient.FFS.WatchJobs(ctx, ch, jid); err != nil {
+		return true, fmt.Sprintf("watching current job %s for bucket %s: %s", jid, buckKey, err), nil
+	}
+
+	var aborted bool
+	var abortMsg string
+	var job ffs.Job
+	select {
+	case <-ctx.Done():
+		log.Infof("job %s status watching canceled", jid)
+		return true, "watching cancelled", nil
+	case s, ok := <-ch:
+		if !ok {
+			return true, "powergate closed communication chan", nil
+		}
+		if s.Err != nil {
+			log.Errorf("job %s update: %s", jid, s.Err)
+			aborted = true
+			abortMsg = s.Err.Error()
+		}
+		job = s.Job
+	}
+
+	if !isJobStatusFinal(job.Status) {
+		return true, "no final status yet", nil
+	}
+
+	// Step 2: On success, save Deal data in the underlying Bucket thread. On
+	// failure save the error message. Also update status on Mongo for the archive.
+	if job.Status == ffs.Success {
+		if err := t.saveDealsInArchive(ctx, buckKey, dbID, dbToken, ffsi.FFSToken, bucketRoot); err != nil {
+			return true, fmt.Sprintf("saving deal data in archive: %s", err), nil
+		}
+	}
+	if err := t.updateArchiveStatus(ctx, ffsi, job, aborted, abortMsg); err != nil {
+		return true, fmt.Sprintf("updating archive status: %s", err), nil
+	}
+
+	msg := "reached final status"
+	if aborted {
+		msg = "aborted with reason " + abortMsg
+	}
+
+	return false, msg, nil
```

Comment:
Say to _not_ reschedule because reached final status (being good or bad, nothing will change that by retrying).